### DISCUSSION TITLE
基于英文文档，实现自动同步

### DISCUSSION THREAD

**Discussion by QC-L** [vuejs/docs-next-zh-cn#522] [Ideas]
Hi, @veaba @Justineo @Jinjiang  我是 QiChang, 目前对 Vite 进行了文档同步的相关改造

手动同步文档的方案，存在一些缺陷。

1. 可能会丢失一些很细节的修改
2. 需要人力去定期同步，耗费精力

我觉得我们可以接入机器人来实现此操作，此方案我已经在 slidev、vitejs、webpack 等文档有了很好效果。

此方案，同样适用于其他语言的文档，我在 slidev 项目中已经实现了为法语和西班牙语实现文档同步。

我可以协助完成此效果，以减轻 @veaba 和 @Jinjiang 的压力。

**Comment by Justineo** [comment]
@QC-L Hi，如 VueConf 当面沟通，我觉得有这个工具对目前的官网文档翻译是个很好的补充。是否方便介绍一下，接入需要做哪些工作，以及后续的更新流程会是什么样的？

**Reply by QC-L** [reply]
好的，我来介绍下。

TODO：

1. 要基于英文重新拉个分支，需要保留英文的 history
2. 将中文内容同步迁移到新拉的分支中，同时进行校对（手动同步的过程中，可能存在遗漏的情况）
	- 我会带着 vite 那边的小伙伴，一起来帮忙
3. 我会基于此分支，让 bot 实现自动同步

具体操作：

由于 Vue 有版本发布的概念，比如 3.2 的内容，就不应该在 3.1 文档中出现，故有以下操作：

1. 分支分为 dev 和 main 分支, 具体请参见 [vitejs/docs-cn](https://github.com/vitejs/docs-cn)
	  - dev 实时与英文文档实现同步
	  - main 分支伴随 Vue 实现发版同步
2. bot 会基于 dev 实现每天/每周同步，将同步频率提高。每次可能只会修改一点点，减少维护者的负担。slidev 目前是每天进行同步。
3. bot 会根据同步内容来决定合并方式，如果有内容冲突，会生成 PR；如果没有内容冲突，会自动 merge
	  - [自动 merge](https://github.com/slidevjs/docs-cn/commit/86f2e584c07a5f6e19ac4710ca8b5840f98878cd)
	  - [内容冲突, 生成 PR](https://github.com/slidevjs/docs-cn/pull/50)

以上就是大概的操作流程，不清楚的地方，可以 @我，我再展开讲解下~

**Reply by Jinjiang** [reply]
了解了，看上去蛮好的！而且我们确实在文档翻译和维护的时候存在相关的痛点。
不介意的话我觉得可以先创建一个个人仓库作为同步的目标仓库“dry-run”几次这个流程，然后我们更具体的评估一下。
谢谢

**Reply by QC-L** [reply]
@Jinjiang 可以的，我基于此文档，进行一些修改，然后 "dry-run" 一段时间，给你们看效果。

**Reply by QC-L** [reply]
一个简单的示例，昨晚 Vue 发布了 3.1.0 版本，新增/修改了很多文档。

假设仓库一直在维护（我故意贴了几个 3.1.0 中有修改的文档）：

1. 因为内容间有冲突，自动生成了[PR](https://github.com/docschina/vue-docs-next/pull/5)
2. 如果没冲突，就会自动合并

![image](https://user-images.githubusercontent.com/13861040/121113109-0d376e80-c844-11eb-8838-96a4b38b2617.png)

上图中为冲突的文件，会把英文修改的地方列出来，我们按照新内容修改即可。

**Comment by QC-L** [comment]
关于锚点的问题，也可以通过 Vite 的方式一同解决

@Jinjiang Vite 目前的解决方案是在标题后添加 

`中文标题 {#英文标题对应的锚点}`

然后读取处理锚点时，获取英文标题对应的锚点信息，进行跳转，可以达到与英文一致的情况

例如：

英文：

[Overview](https://vitejs.dev/guide/#overview) - https://vitejs.dev/guide/#overview

中文：

[总览](https://cn.vitejs.dev/guide/#overview) - https://cn.vitejs.dev/guide/#overview

**Comment by Otto-J** [comment]
自动同步的工具有进展了吗，vue3.2新增的内容目前还没有同步
